### DISCUSSION TITLE
restarting an experiment without reloading the page

### DISCUSSION THREAD

**Discussion by Ariana-Mg** [jspsych/jsPsych#1444] [Q&A]
Hi,

I have made a gamified experiment on jspsych and want to give participants the opportunity to restart it after they have finished it. To restart the experiment, I want to present participants with a restart button at the end. 

I want their decision to restart the game to be recorded as part of data collection. I also want their data from previous run-throughs to be saved.  

Any help would be very much appreciated!

Thanks!

**Comment by becky-gilbert** [comment]
Hi @Ariana-Mg, I'm not sure if this is the best/only solution, but you could put your entire game timeline into a [looping timeline](https://www.jspsych.org/overview/timeline/#looping-timelines). The last trial on the game timeline would contain the reset button. The loop function runs after the whole `timeline` array finishes, so inside that function you could check to see whether the participant clicked the restart button on the last trial - if so then repeat the game timeline (`return true`), if not then don't repeat the game timeline (`return false`). All data from each timeline repetition is saved to the jsPsych data by default.
